src/library.cpp

```diff
@@ -331,7 +331,7 @@ Library::BookIdCollection Library::filter(const Filter& filter)
 {
   BookIdCollection result;
   for(auto id : getBooksByTitleOrDescription(filter)) {
-    if(filter.acceptByNonQueryCriteria(m_books[id])) {
```

Comment:
This bug was introduced in #460 because `Library::filter()` [is not declared `const`](https://github.com/kiwix/kiwix-lib/issues/487).